lighthouse-core/gather/driver.js

```diff
@@ -85,12 +75,25 @@ class Driver {
    */
   constructor(connection) {
     this._connection = connection;
-    this._networkMonitor = new NetworkMonitor(this);
 
     this.on('Target.attachedToTarget', event => {
       this._handleTargetAttached(event).catch(this._handleEventError);
     });
 
+    this.on('Page.frameNavigated', event => {
+      // We're only interested in setting autoattach on the root via this method.
+      // `_handleTargetAttached` takes care of the recursive piece.
+      if (event.frame.parentId) return;
+
+      // Enable auto-attaching to subtargets so we receive iframe information.
+      this.sendCommand('Target.setAutoAttach', {
+        flatten: true,
+        autoAttach: true,
+        // Pause targets on startup so we don't miss anything
+        waitForDebuggerOnStart: true,
+      }).catch(this._handleEventError);
+    });
```

Comment:
Why move this up here? Wouldn't auto-attach remain enabled for any child frames after this?

Reply:
We want to enable autoattach for the current root frame session. This was previously done in `gotoURL`, but pretty unrelated to the process of navigating. I don't see a reason this couldn't be done just once in `setupDriver` but for consistency sake I'm matching the previous behavior of every navigation as this will be redone differently in FR puppeteer land anyway.

> Wouldn't auto-attach remain enabled for any child frames after this?

Is this pointing out what I said above about "I don't see a reason this couldn't be done just once"? :)

Reply:
> Is this pointing out what I said above about "I don't see a reason this couldn't be done just once"?

I was saying that if we leave auto-attach on after the main frame navigation, would we auto-attach to any child frames as well?